pallets/cash/src/core.rs

```diff
@@ -1226,20 +1231,31 @@ pub fn on_initialize_internal<T: Config>(
     if now == 0 {
         return Err(Reason::TimeTravelNotAllowed);
     }
-    if last_yield_timestamp == 0 {
-        // this is the first time we have seen a valid time, set it
-        LastYieldTimestamp::put(now);
+    if last_yield_timestamp == 0 || last_block_timestamp == 0 {
```

Comment:
dev note - this is something of a belt and suspenders issue - last_yield_timestamp is actually meant to be set by `config()` / `chainSpec` but having a safegaurd here may not be the worst thing? It was here before, i just updated it

Reply:
Never heard that term before, I like it ;) Not sure about the right thing here overall yet but I think this is good now